x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/AggregationResultUtils.java

```diff
@@ -466,22 +452,14 @@ public Object value(Aggregation aggregation, Map<String, String> fieldTypeMap, S
         @Override
         public Object value(Object key, String type) {
             assert key instanceof String;
-            Rectangle rectangle = GeoTileUtils.toBoundingBox(key.toString());
-            final Map<String, Object> geoShape = new HashMap<>();
-            geoShape.put(ShapeParser.FIELD_TYPE.getPreferredName(), PolygonBuilder.TYPE.shapeName());
-            geoShape.put(
-                ShapeParser.FIELD_COORDINATES.getPreferredName(),
-                Collections.singletonList(
-                    Arrays.asList(
-                        new Double[] { rectangle.getMaxLon(), rectangle.getMinLat() },
-                        new Double[] { rectangle.getMinLon(), rectangle.getMinLat() },
-                        new Double[] { rectangle.getMinLon(), rectangle.getMaxLat() },
-                        new Double[] { rectangle.getMaxLon(), rectangle.getMaxLat() },
-                        new Double[] { rectangle.getMaxLon(), rectangle.getMinLat() }
-                    )
+            Rectangle r = GeoTileUtils.toBoundingBox(key.toString());
+            Polygon polygon = new Polygon(
+                new LinearRing(
+                    new double[]{r.getMinLon(), r.getMaxLon(), r.getMaxLon(), r.getMinLon(), r.getMinLon()},
```

Comment:
Does the order matter here? It seems that this 2-D array is *not* a transposition of the 2-D array on the left-hand side.

Reply:
Probably it does no matter because they represent the same polygon.